File: examples/sicora.py
#
# jython examples for jas.
# $Id$
#

import sys;

from jas import PolyRing, ZZ, Order, Ideal
from jas import startLog, terminate

# sicora, e-gb example

r = PolyRing( ZZ(), "t", Order.INVLEX );
print "Ring: " + str(r);
print;

f1 = 2 * t + 1;
f2 = t**2 + 1;

F = r.ideal( "", [f1,f2] );
print "Ideal: " + str(F);
print;

E = F.eGB();
print "seq e-GB:", E;
print "is e-GB:", E.iseGB();
print;

f = t**3;
n = E.eReduction(f);
print "e-Reduction = " + str(n);

File: examples/sicora.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# sicora, e-gb example

r = PolyRing.new(ZZ(),"t",Order::INVLEX)
puts "Ring: " + str(r);
puts;
#automatic: one,t = r.gens();

f1 = 2 * t + 1;
f2 = t**2 + 1;

ff = r.ideal( "", [f1,f2] );
puts "ideal: " + str(ff);
puts;

gg = ff.eGB();
puts "seq e-GB: " + str(gg);
puts "is e-GB: " + str(gg.iseGB());
puts;

f = t**3;
n = gg.eReduction(f);
puts "e-Reduction = " + str(n);

